### ISSUE TITLE
Build failed when using NotifierProvider.family

### ISSUE THREAD

**Issue by khoadng** [rrousselGit/riverpod#2536]
**Describe the bug**
Can't build when using NotifierProvider.family. No compile errors.

This error only happen during the build phase.
```
Error: Type argument 'SomeNotifier' doesn't conform to the bound 'FamilyNotifier<State, Arg>' of the type variable 'NotifierT' on 'NotifierProviderFamilyBuilder.call'.
```

**To Reproduce**
```dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

final aProviderFamily =
    NotifierProvider.family<SomeNotifier, int, int>(() => SomeNotifier());

class SomeNotifier extends AutoDisposeNotifier<int> {
  SomeNotifier() : super();

  @override
  int build() => 0;

  void increment() => state++;
}

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: Test(),
    );
  }
}

class Test extends ConsumerWidget {
  const Test({super.key});
  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final count = ref.watch(aProviderFamily(0));
    return Scaffold(
      body: Text(count.toString()),
    );
  }
}

```

**Expected behavior**
It should work, or at least there should be some compile errors.

**Comment by rrousselGit** [comment]
There is a compile error, hence why your build is failing.

The issue is the analyzer not showing the compile error.
It's a dart bug I reported before.


TL;DR, you subclassed the wrong notifier. Use "extends FamilyNotifier" in your case.
